### PR TITLE
Draft: LISA + SAM integration

### PR DESCRIPTION
**Pull request by NVShreyas**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py [hunk 1]
```diff
@@ -1199,6 +1218,8 @@ def sharded_state_dict(self, prefix: str = ''):
 
         original_sharded_state_dict = super().sharded_state_dict()
         keys_to_keep = self.get_keys_to_keep()
+        #TODO: this breaks. It can't find this tensor in the sharded dict, probably the reason why its not loaded.
+        # keys_to_keep.append("model.lisa_sam.sam.prompt_encoder.pe_layer.positional_encoding_gaussian_matrix")
```
Comment (NVShreyas) [inline nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py:1222]: @anmolgupt to fix prompt encoder get dense PE, I believe we have to make it sharded so that it can read the tensor from the checkpoint. It's in the checkpoint, but it's not loaded due to this.
